src/ImageProcessor.Web/HttpModules/ImageProcessingModule.cs

```diff
@@ -826,7 +826,10 @@ private IImageService GetImageServiceForRequest(string url, string applicationPa
             foreach (IImageService service in services)
             {
                 string key = service.Prefix;
-                if (!string.IsNullOrWhiteSpace(key) && path.StartsWith(key, StringComparison.InvariantCultureIgnoreCase))
+                if (!string.IsNullOrWhiteSpace(key) 
+                    && (service.IsRegex 
```

Comment:
Bugger, you're correct! It' looks like we'll have to delay this then as I cannot add the property until I release V5. Shame!